sdk/sdkservices/sessions.go

```diff
@@ -13,12 +13,22 @@ type ListSessionsFilter struct {
 	BuildID      sdktypes.BuildID
 	StateType    sdktypes.SessionStateType
 	CountOnly    bool
+
+	PageSize  int32
+	Skip      int32
+	PageToken string
+}
+
+type ListSessionResult struct {
+	Sessions      []sdktypes.Session
+	Count         int
+	NextPageToken string
 }
 
 type Sessions interface {
 	Start(ctx context.Context, session sdktypes.Session) (sdktypes.SessionID, error)
 	Stop(ctx context.Context, sessionID sdktypes.SessionID, reason string, force bool) error
-	List(ctx context.Context, filter ListSessionsFilter) ([]sdktypes.Session, int, error)
+	List(ctx context.Context, filter ListSessionsFilter) (ListSessionResult, error)
```

Comment:
```suggestion
	List(ctx context.Context, filter ListSessionsFilter) ([]sdktypes.Session, PaginationInfo, error)
```

then you could reuse this everywhere.

Reply:
WDYM ? why can't I reuse it everywhere anyway?

Reply:
I did some changes, 
created an embeddable struct 
PTAL and let me know if that makes sense for you

in any case, let's ack this PR and open another ticket to change it if we want a different design

Reply:
i still think these should be separate return vals, no need for the results struct.

I'll approve, but please fix that.